xcdat/dataset.py

```diff
@@ -0,0 +1,43 @@
+"""Dataset module for Dataset related functions."""
+import xarray as xr
+
+from xcdat import axis  # noqa: F401
+
+
+def open_dataset(path: str) -> xr.Dataset:
+    """Wrapper for xarray's ``open_dataset()`` to apply common operations.
+
+    Operations include:
+
+    - Generating axis bounds if they don't exist
+
+    Parameters
+    ----------
+    path : str
+        Path to Dataset.
+
+    Returns
+    -------
+    xr.Dataset
+        Dataset after applying operations.
+
+    Notes
+    -----
+    xarray's ``open_dataset`` opens the file with read-only access. When you
+    modify values of a Dataset, even one linked to files on disk, only the
+    in-memory copy you are manipulating in xarray is modified: the original file
+    on disk is never touched.
+
+    Examples
+    --------
+    Import and call module:
+
+    >>> from xcdat import open_dataset
+    >>> ds = open_dataset("file_path")
+    """
+    ds = xr.open_dataset(path)
```

Comment:
@tomvothecoder - One xarray feature that I expect we will frequently rely on is the [open_mfdataset](http://xarray.pydata.org/en/stable/generated/xarray.open_mfdataset.html) functionality. I'm wondering if we could automatically decide whether to use `open_dataset` or `open_mfdataset` by checking if the `path` parameter is a file or a directory, e.g.,

```
import os
...
if os.path.isdir(path):
    ds = xr.open_mfdataset(path + '*')
elif os.path.isfile(path):
    ds = xr.open_dataset(path)
else:
    raise ValueError('Input path (' + path + ') is invalid.')
```